analyzers/tests/SonarAnalyzer.TestFramework/Verification/IssueValidation/IssueLocationPair.cs

```diff
@@ -0,0 +1,87 @@
+﻿/*
+ * SonarAnalyzer for .NET
+ * Copyright (C) 2015-2024 SonarSource SA
+ * mailto: contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System.Text;
+
+namespace SonarAnalyzer.TestFramework.Verification.IssueValidation;
+
+internal sealed record IssueLocationPair(IssueLocation Actual, IssueLocation Expected)
+{
+    public string FilePath => Actual?.FilePath ?? Expected.FilePath;
+    public int LineNumber => Actual?.LineNumber ?? Expected.LineNumber;
+    public int? Start => Actual?.Start ?? Expected.Start;
+
```

Comment:
No idea just asking - is is possible to have an a half-defined actual issue and get mixed assignments to these fields?
for example filepath will be from actual and line number from expected.

Reply:
The pair is always created from issues that matches on file and line. So these will always have correct value.
There's a slight nuance in project-level issues that do not have any location. There, the Expected will win. This will be solved later.